analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LoopsAndLinq.cs

```diff
@@ -169,6 +177,15 @@ public void ForEachWithPropertyGet_SuggestsSelect(ICollection<string> collection
                 }
             }
 
+            foreach (var element in collection) // Noncompliant
```

Comment:
looking at this better - wouldn't a secondary location be helpful, pointing at the `if` ? to have a visual helper that "hey, I am raising on the `foreach` and here's an `if` which can be moved in the foreach"

IMO this will help the developer see how to fix the issue without opening the rule description

Reply:
Indeed.

Reply:
I've added secondary locations for Where suggestions. For Select(), I think this is not needed since the property can be used in multiple places and we already include the property name in the error message.